pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-hadoop/src/main/java/org/apache/pinot/plugin/ingestion/batch/hadoop/HadoopSegmentGenerationJobRunner.java

```diff
@@ -132,7 +132,7 @@ public void run()
     //init all file systems
     List<PinotFSSpec> pinotFSSpecs = _spec.getPinotFSSpecs();
     for (PinotFSSpec pinotFSSpec : pinotFSSpecs) {
-      new PinotConfiguration(pinotFSSpec);
+      PinotFSFactory.register(pinotFSSpec.getScheme(), pinotFSSpec.getClassName(), new PinotConfiguration(pinotFSSpec));
```

Comment:
can we delete the constructor from PinotConfiguration

```  
public PinotConfiguration(PinotFSSpec pinotFSSpec) {
      }
```

Reply:
Considering that it only copies properties from `PinotFSSpec.getConfig`, it seems like a reasonable ask.

Reply:
So, here's my second thoughts about this. `PinotFSSpec.getConfig` returns a `Map<String, String>` where the the other constructor takes `Map<String, Object>`. This use case only occurs when `PinotFSSpec` is involve. If I remove the constructor, I will have to setup an abstract utilitary function that translates the `Map<String, String>` to `Map<String, Object>` anyway.

Reply:
Few other options
- add a constructor PinotConfiguration(map<string,string> )
- change getConfigs() in pinotFSSpec to return map<string,object>

Reply:
PinotConfiguration(map<string,string>) will conflict with the existing `PinotConfiguration(map<string,object>)` What I can do is rewrite `PinotConfiguration(map<string,string>)` as a static function `PinotConfiguation.fromStringProperties(map<string,object>)` and have that function include the mapping logic of the current `PinotConfiguration(PinotFSSpec pinotFSSpec)`. `PinotFSSpec` would be out of the way with that.

Reply:
sure

Reply:
you can do this in the next PR